### DISCUSSION TITLE
How to render a KaTeX environment

### DISCUSSION THREAD

**Discussion by pirivan** [jpanther/congo#586] [Q&A] [answered]
Hello,

I'm trying to render one of the KaTeX [supported environments](https://katex.org/docs/supported.html#environments) in an article (post) but I can't get it to work. Here is an example using one of the samples in the KaTeX documentation, which using `congo` just renders as literal text:

```text
{{< katex >}}
$$
x = \begin{cases}
   a &\text{if } b \\
   c &\text{if } d
\end{cases}
$$
```
 
In fact, I can't get `congo` to render any `$$...$$` blocks that include any `\begin{something} ... \end{something}` syntax. Note that rendering other math works fine, for example, `$$\frac{1}{2}$$`. I'm using `congo v2.6.1` with Hugo v0.113.0+extended.

Any suggestions?

Thank you

**Comment by jpanther** [comment]
I'm not a KaTeX expert but is this as simple as using `\cr` instead of `\\` in your markup?

**Reply by pirivan** [reply, marked as answer]
Hmm, yes, or by using triple `\\\` to mark the line breaks, as I just found out!
Problem solved, thank you for answer!

**Reply by jpanther** [reply]
I think it stems from limitations around how Hugo processes Markdown text and so certain character sequences need to be escaped. Glad it solved your issue.
